### PR TITLE
Document strictly-increasing requirement for RegularGridInterpolator

### LINKED ISSUE #36499: The `RegularGridInterpolator` requires that the points value be strictly **increasing**, not merely monotonic, not specified in docs and differs from scipy
### Description

The function can be found here:
https://github.com/jax-ml/jax/blob/753f089dad98728d6444f732323e020f109fec0d/jax/_src/third_party/scipy/interpolate.py#L36-L163

The docstring includes this example which works:
```python
points = (jnp.array([1, 2, 3]), jnp.array([4, 5, 6]))
values = jnp.array([[10, 20, 30], [40, 50, 60], [70, 80, 90]])
interpolate = RegularGridInterpolator(points, values, method='linear')
query_points = jnp.array([[1.5, 4.5], [2.2, 5.8]])
interpolate(query_points)
```
returns `Out[8]: Array([30., 64.], dtype=float32)`

However, simply reversing on of the axis directions results in all outputs as NaN:
```python
points = (jnp.array([1, 2, 3]), jnp.array([6, 5, 4]))
values = jnp.array([[10, 20, 30], [40, 50, 60], [70, 80, 90]])
interpolate = RegularGridInterpolator(points, values, method='linear')
query_points = jnp.array([[1.5, 4.5], [2.2, 5.8]])
interpolate(query_points)
```
returns `Out[19]: Array([nan, nan], dtype=float32)`

My workaround fix is to reorder things before creation of the jax objects so that the axis values are increasing.

This behavior differs from SciPy's implementation which only requires that the axis values be monotonic, so this doesn't serve as a direct drop in replacement without a fix. I think it can be addressed with some `jnp.flip` options internally. I'm happy to work on a PR if this is of interest. At the very least I would like to update the documentation to explain the difference.

### System info (python version, jaxlib version, accelerator, etc.)

In [25]: jax.print_environment_info()
jax:    0.9.2
jaxlib: 0.9.2
numpy:  2.4.2
python: 3.12.8 (main, Jan 11 2025, 09:42:09) [GCC 14.2.0]
device info: cpu-1, 1 local devices"
process_count: 1
platform: uname_result(system='Linux', node='none', release='6.17.0-14-generic', version='#14~24.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Jan 15 15:52:10 UTC 2', machine='x86_64')

### PR DESCRIPTION
**Pull request by adityasingh2400**
## Summary

`jax.scipy.interpolate.RegularGridInterpolator` requires each axis of `points` to be **strictly increasing**, but `scipy.interpolate.RegularGridInterpolator` accepts any monotonic axis (increasing or decreasing). The mismatch is surprising and the failure mode is opaque: a reversed axis silently produces all-`NaN` outputs from the interpolation rather than raising a clear error.

This PR addresses both halves of the gap reported in #36499:

- **Docstring `Note:` section** explicitly calls out the difference from SciPy and points users at the fix (reorder the axis and the corresponding axis of `values`).
- **Construction-time validation** raises a precise `ValueError` when an axis is not strictly increasing (e.g. reversed or with repeated values), so users get a clear refusal instead of downstream `NaN`s. The check is concrete-only (skipped under tracing via `core.is_concrete`), which preserves the existing JIT and pytree round-trip behavior. The pre-existing TODO about JIT-able sanity checks is left in place.
- **Tests** cover the strictly-increasing happy path plus reversed-axis and repeated-axis rejection.
- **CHANGELOG** entry under "Breaking changes" notes the new construction-time error, since code that previously got silent `NaN`s on reversed axes will now see a `ValueError`.

This intentionally does not pursue the deeper "make decreasing axes work via internal `jnp.flip`" scope mentioned in the issue; the docstring + early error is a strictly better state and unblocks users today.

Fixes #36499

## Test plan

- [x] `pytest tests/scipy_interpolate_test.py -v` (4 existing parametrized tests + 3 new tests all pass)
- [x] `ruff check jax/_src/third_party/scipy/interpolate.py tests/scipy_interpolate_test.py` clean
- [x] Manual verification that the docstring example still runs and that `RegularGridInterpolator` instances continue to round-trip through `jax.jit` via the existing `register_pytree_node` registration (the validation is skipped on tracers).

### CODE REVIEW

**Comment by adityasingh2400** [comment]
@googlebot I signed it!

**Comment by adityasingh2400** [comment]
Thanks for the suggestion. Hoisted the rank check out of the concrete-only branch so it fires during tracing too, kept the strictly-increasing check inside the concrete branch since it needs values. Force-pushed as a95f0ef.

File: jax/_src/third_party/scipy/interpolate.py [hunk 1]
```diff
@@ -93,6 +102,27 @@ def __init__(self,
...
+    # SciPy accepts any monotonic axis, but this implementation only handles
+    # strictly increasing axes (a decreasing axis silently produces NaNs from
+    # the interpolation). The rank check runs unconditionally since shape is
+    # known statically; the strictly-increasing check requires concrete values
+    # and is skipped under tracing.
+    for axis, g in enumerate(self.grid):
+      if g.ndim != 1:
+        raise ValueError(
+            f"points[{axis}] must be a 1-D array; got shape {g.shape}")
+      if not core.is_concrete(g):
```
Comment (jakevdp) [inline jax/_src/third_party/scipy/interpolate.py]: Please remove lines 114-125, and update the documentation to match. We should not do static runtime checks, because having a validation check that does not execute under JIT or other transformations can be surprising and misleading.

**Review by jakevdp — CHANGES_REQUESTED** [review]

**Comment by adityasingh2400** [comment]
Done in 8d62a2f. I dropped the runtime check block at lines 114-125 and softened the docstring note so it no longer mentions construction-time validation. The only remaining change in the PR is the docstring note in jax/_src/third_party/scipy/interpolate.py recording that each axis of points must be strictly increasing.

**Review by jakevdp — APPROVED** [review]
Thanks!
